pkg/controller/pod_volume_backup_controller.go

```diff
@@ -18,330 +18,220 @@ package controller
+	pvb.Status.StartTimestamp = &metav1.Time{Time: r.Clock.Now()}
 
-	// if there's a caCert on the ObjectStorage, write it to disk so that it can be passed to restic
-	var caCertFile string
-	if backupLocation.Spec.ObjectStorage != nil && backupLocation.Spec.ObjectStorage.CACert != nil {
-		caCertFile, err = restic.TempCACertFile(backupLocation.Spec.ObjectStorage.CACert, req.Spec.BackupStorageLocation, c.fileSystem)
-		if err != nil {
-			log.WithError(err).Error("Error creating temp cacert file")
-		}
-		// ignore error since there's nothing we can do and it's a temp file.
-		defer os.Remove(caCertFile)
+	pod := corev1.Pod{}
+	if err := r.Client.Get(r.Ctx, req.NamespacedName, &pod); err != nil {
+		return r.logErrorAndUpdateStatus(
+			log,
+			&pvb,
+			err,
+			fmt.Sprintf("getting pod %s/%s", pvb.Spec.Pod.Namespace, pvb.Spec.Pod.Name),
+		)
 	}
-	resticCmd.CACertFile = caCertFile
 
-	env, err := restic.CmdEnv(backupLocation, c.credentialsFileStore)
+	var resticDetails resticDetails
+	resticCmd, err := r.buildResticCommand(log, req.Namespace, &pvb, &pod, &resticDetails)
 	if err != nil {
-		return c.fail(req, errors.Wrap(err, "error setting restic cmd env").Error(), log)
-	}
-	resticCmd.Env = env
-
-	// If this is a PVC, look for the most recent completed pod volume backup for it and get
-	// its restic snapshot ID to use as the value of the `--parent` flag. Without this,
-	// if the pod using the PVC (and therefore the directory path under /host_pods/) has
-	// changed since the PVC's last backup, restic will not be able to identify a suitable
-	// parent snapshot to use, and will have to do a full rescan of the contents of the PVC.
-	if pvcUID, ok := req.Labels[velerov1api.PVCUIDLabel]; ok {
-		parentSnapshotID := getParentSnapshot(log, pvcUID, req.Spec.BackupStorageLocation, c.podVolumeBackupLister.PodVolumeBackups(req.Namespace))
-		if parentSnapshotID == "" {
-			log.Info("No parent snapshot found for PVC, not using --parent flag for this backup")
-		} else {
-			log.WithField("parentSnapshotID", parentSnapshotID).Info("Setting --parent flag for this backup")
-			resticCmd.ExtraFlags = append(resticCmd.ExtraFlags, fmt.Sprintf("--parent=%s", parentSnapshotID))
-		}
+		return r.logErrorAndUpdateStatus(log, &pvb, err, "building restic command")
 	}
 
-	var stdout, stderr string
-
 	var emptySnapshot bool
-	if stdout, stderr, err = restic.RunBackup(resticCmd, log, c.updateBackupProgressFunc(req, log)); err != nil {
+	stdout, stderr, err := r.ResticExec.RunBackup(resticCmd, log, r.updateBackupProgressFunc(&pvb, log))
+	if err != nil {
 		if strings.Contains(stderr, "snapshot is empty") {
 			emptySnapshot = true
 		} else {
-			log.WithError(errors.WithStack(err)).Errorf("Error running command=%s, stdout=%s, stderr=%s", resticCmd.String(), stdout, stderr)
-			return c.fail(req, fmt.Sprintf("error running restic backup, stderr=%s: %s", stderr, err.Error()), log)
+			log.WithError(errors.WithStack(err)).Errorf(
```

Comment:
I think this error log can be removed. 
The return error can be logged in controller-manager logger.

Reply:
Yes, I think you're right. When errors are returned, I think they'll be printed in calling functions. They don't need to be logged again.

Reply:
Errors that are returned are no longer logged now.